### ISSUE TITLE
Simple framework kernel for anisotropic heat conduction 

### ISSUE THREAD

**Issue by snschune** [idaholab/moose#21832]
## Reason

This is supported using `PINSFVEnergyAnisotropicDiffusion` but a simple kernel in the framework
would be nice

## Design

Add a new kernel that takes a vector diffusion coefficient (diagonal tensor).

## Impact

None expected

**Comment by dschwen** [comment]
This doesn't work for you?
https://github.com/idaholab/moose/blob/next/modules/phase_field/doc/content/source/kernels/MatAnisoDiffusion.md

**Comment by dschwen** [comment]
(or `AnisoHeatConduction`)

**Comment by snschune** [comment]
@lindsayad  
In `FVDiffusion`, is 

```
const auto k = _coeff(Moose::FV::makeCDFace(*_face_info, faceArgSubdomains()));
```
essentially doing:
```
k_f = g * k_C + (1 - g) * k_N?
```
If so, I think we can do better than that. Should be:
```
1/k_f = g / k_C + (1 - g) / k_N.
```
see Moukalled p 226.

**Comment by dschwen** [comment]
Ugh, I missed the FV part. I'll delete my comments in shame...

**Comment by grmnptr** [comment]
Okay, I explored a little bit this topic yesterday. I took a 2D verification case from here (with slight modifications because they have typos in there):
 
```
Stefano Ferraris, Ivan Bevilacqua, Davide Canone, Davide Pognant, Maurizio Previati, "The Finite Volume Formulation for 2D Second-Order Elliptic Problems with Discontinuous Diffusion/Dispersion Coefficients", Mathematical Problems in Engineering, vol. 2012, Article ID 187634, 23 pages, 2012. https://doi.org/10.1155/2012/187634
```

So the domain is the following: `\Omega = [0,1]x[0,1]` with `domain 1` being `y<0.5` and `domain 2` what is left. The mesh is below. I used regular triangles because they will require nonorthogonal correction but we don't require skewness-correction. The diffusivity in `domain 1` and `domain 2` are `d1` and `d2` respectively. I used d1=1 and d2=10 in this case, I think that is a considerable difference in diffusivities.

The exact solution is constructed to be continuous with a continuous normal flux on the interface. It is also constructed to enable 2D effects.

I explored four approaches:
1. The current approach we have in MOOSE (approximate the face gradient using an orthogonal and a nonorthogonal component and interpolate the face diffusivity using `linear interpolation`)
2. Approximate the face gradient using an orthogonal and a nonorthogonal component and interpolate the face diffusivity using `harmonic interpolation`
3. Compute cell gradients only and compute the face fluxes by linearly interpolating k*grad(u)|cell to the faces and using the current approach in MOOSE to compute the face gradient on the boundary faces.

I created a python script to do the MMS case for all 4 approaches. I also double checked with MOOSE results for (1) and (2) and I got matching results. (3) comes from the python script only.

The results for the MMS case

```
refinement | case (1) | case (2) | case (3) |
0 | 5.81E-2  | 3.69E-2 | 8.64E-2
1 | 4.06E-2   | 1.30E-2 | 5.12E-2
2 | 2.30E-2   | 3.81E-3 | 2.61E-2
3 | 1.19E-2  | 1.08E-3 | 1.29E-2
4 | 6.04E-3  | 3.06E-4 | 6.41E-3
```

So We see that interpolating a flux results in higher errors than the current approach in MOOSE, but the harmonic interpolation for the diffusivity to the face would be considerably more accurate. So I think we could add the harmonic interpolation as an option for the user. I think the increased accuracy (still not entirely 2nd order) comes from the derivation in Moukalled considering that in the mesh above most of the face normal gradient estimation comes from the orthogonal component which is almost identical to the terms described in his approach. Looks like the nonorthogonal-corrector has a moderate effect in this case.

<img width="1845" alt="Screen Shot 2022-08-19 at 9 08 27 A
<img width="698" alt="Screen Shot 2022-08-19 at 9 09 38 AM" src="https://user-images.githubusercontent.com/31662443/185649895-22b616b3-1995-4d0b-8c4c-71fedd75933c.png">
M" src="https://user-images.githubusercontent.com/31662443/185649739-acd76cbb-952d-4cab-9f76-ab289d547a5a.png">
